lib/galaxy/webapps/base/webapp.py

```diff
@@ -951,8 +951,8 @@ def get_or_create_default_history(self):
             if default_history:
                 history = default_history
                 self.set_history(history)
-        else:
-            history = self.new_history()
+            else:
+                history = self.new_history()
```

Comment:
```suggestion
            else:
                history = self.new_history()
         else:
             history = self.new_history()
```

Reply:
I guess we need both ... I'll commit but feel free to drop if that's not the right thing.

Reply:
See, I knew I wanted to look at this again in the morning :)  Yeah, we need both cases; I'm going to just refactor this a little and let's get it in.